targets/TARGET_STM/TARGET_STM32F4/STM32Cube_FW/CMakeLists.txt

```diff
@@ -1,15 +1,107 @@
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_ltdc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_ltdc_ex.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_mmc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_nand.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_nor.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_pccard.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_pcd.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_pcd_ex.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_pwr.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_pwr_ex.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_qspi.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_rcc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_rcc_ex.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_rng.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_rtc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_rtc_ex.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_sai.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_sai_ex.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_sd.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_sdram.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_smartcard.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_smbus.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_spdifrx.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_spi.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_sram.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_tim.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_tim_ex.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_uart.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_usart.c
+        STM32F4xx_HAL_Driver/stm32f4xx_hal_wwdg.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_adc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_crc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_dac.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_dma.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_dma2d.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_exti.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_fmc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_fmpi2c.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_fsmc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_gpio.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_i2c.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_lptim.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_pwr.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_rcc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_rng.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_rtc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_sdmmc.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_spi.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_tim.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_usart.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_usb.c
+        STM32F4xx_HAL_Driver/stm32f4xx_ll_utils.c
+
+        STM32F4xx_HAL_Driver/Legacy/stm32f4xx_hal_can_legacy.c
 )
 
 target_include_directories(mbed-core
     INTERFACE
         ${CMAKE_CURRENT_SOURCE_DIR}
         ${CMAKE_CURRENT_SOURCE_DIR}/CMSIS
+        ${CMAKE_CURRENT_SOURCE_DIR}/STM32F4xx_HAL_Driver
+        ${CMAKE_CURRENT_SOURCE_DIR}/STM32F4xx_HAL_Driver/Legacy
```

Comment:
This is the same as here:

https://github.com/ARMmbed/mbed-os/pull/13835/files#diff-7c98b07952b498f1b0449bd8fdc3c0f1c2dac26ff61884d00a56c8aa8af36e41R26-R29

I think it can be removed from the TARGET_STM32F4/CMakeLists.txt

Reply:
Is this referring to `CMAKE_CURRENT_SOURCE_DIR` or something else? the diff points to k66f file

Reply:
Yeah sorry, the copy/paste in diff is not working well when you expand the files.

I meant here:

https://github.com/0xc0170/mbed-os/blob/24301485ea6f350ade61eb8f4486f5b2dd7bc0e8/targets/TARGET_STM/TARGET_STM32F4/CMakeLists.txt#L26-L29

```cmake
target_include_directories(mbed-core
    INTERFACE
        ${CMAKE_CURRENT_SOURCE_DIR}
        ${CMAKE_CURRENT_SOURCE_DIR}/STM32Cube_FW
        ${CMAKE_CURRENT_SOURCE_DIR}/STM32Cube_FW/CMSIS
        ${CMAKE_CURRENT_SOURCE_DIR}/STM32Cube_FW/STM32F4xx_HAL_Driver
        ${CMAKE_CURRENT_SOURCE_DIR}/STM32Cube_FW/STM32F4xx_HAL_Driver/Legacy
)
```

is the same as here:

https://github.com/0xc0170/mbed-os/blob/24301485ea6f350ade61eb8f4486f5b2dd7bc0e8/targets/TARGET_STM/TARGET_STM32F4/STM32Cube_FW/CMakeLists.txt#L103-L106

```cmake
target_include_directories(mbed-core
    INTERFACE
        ${CMAKE_CURRENT_SOURCE_DIR}
        ${CMAKE_CURRENT_SOURCE_DIR}/CMSIS
        ${CMAKE_CURRENT_SOURCE_DIR}/STM32F4xx_HAL_Driver
        ${CMAKE_CURRENT_SOURCE_DIR}/STM32F4xx_HAL_Driver/Legacy
)
```

in terms of includes.

Reply:
+1, fixed now